### ISSUE TITLE
Test builds from 0.3.6 to 0.3.8, then use the ansible build for 0.3.9

### ISSUE THREAD

**Issue by infosec-alchemist** [unfetter-discover/unfetter#1329]
Do we need any scripts to update backend schemas or config files to move from 0.3.6 to 0.3.8.

To move from 0.3.8 to 0.3.9 we can use the ansible scripts for migration.

**Comment by infosec-alchemist** [comment]
From the call today, there is a script that has to be run to move from 0.3.6 to 0.3.7.  Thus, this issue should stay open until we identify that script and build the upgrade command

**Comment by infosec-alchemist** [comment]
I'm seeing problems with the upgrade:
- The private-config.json is actually a directory inside the containers.  I am not sure why.  But the backup breaks. 
- The backup failed because it is assuming the private-config.json is inside the config directories rather than the config/private.  so it breaks and never backs up the mongodb
- The hosts.ini file has the registry as "".  Therefore, it doesn't properly pull the docker images from the correct location.
- The 0.3.10 version, with the splits in the group vars might be more managable
- #1392 might really improve the whole process

**Comment by webiator** [comment]
- If you first try to bring up Unfetter without having created the config/private/private-config.json file, docker-compose creates a directory by that name when mapping to the container (I guess it thinks it is being helpful). That is why you have that directory. That directory, of course, is not actually supposed to be there. This is why the UAC configuration script really has to be executed before any attempt to start the stack happens.

- I can't swear to ever having a successful Mongo backup. I'm leaning toward "probably no." Is there a way I can validate the backup? I don't want to update our highside UAC deployment if it means all the customer data will be lost...

- The extra "certs" directory in the path certainly explains that problem.

**Comment by infosec-alchemist** [comment]
Okay, so, the private-config.json directory is created when the docker-compose builds the stack without the api-config script is run?  That makes sense.  

I'm testing the backup by creating a new analytic.  Running the backup.  Shut down everything, delete the old database directory, then bringing up the entirely new stack with Ansible.

**Comment by webiator** [comment]
Before I brought up the test demo instance yesterday, I deleted the mongo volume so the playbook would have to re-create it. After I brought it up, I went inside the container and checked to see if my user record was there, and it was. So what I think happened earlier was that the volume was created, but because of issues with either the certs or the config files, the database was not restored. So now that we are past those other issues, I think the database is okay.

Also, the migration script (for 0.3.8 -> 0.3.9) worked perfectly, so I've feeling pretty hopeful for updating the customer UAC instance.
